tap_linkedin_ads/sync.py

```diff
@@ -547,7 +547,13 @@ def sync_ad_analytics(
 ):
     # pylint: disable=too-many-branches,too-many-statements,unused-argument
 
-    last_datetime = get_bookmark(state, stream_name, last_datetime)
+    # This is a bit of a filthy hack to make the tap respect assigned deltas without modifying too much.
+    # Frankly, the state handling and dependency here is horrid.
+    last_datetime = datetime(
+        static_params["dateRange.start.year"],
+        static_params["dateRange.start.month"],
+        static_params["dateRange.start.day"],
+    ).isoformat()
```

Comment:
does the `static_params` also consider the previous state? From the line removed, it just looks like we are throwing away the state and always running it from 7 days back? What if the last time the tap was run was 20 days ago, and we in fact need to go back 27 days (considering the 7 day lookback) - will we do that, or just go back 7 days?

Reply:
In short: yes.

You can see the origin of the passed in `static_params` [way up in the call chain](https://github.com/dreamdata-io/tap-linkedin-ads/blob/c102e151b060c1f5cbbe37ac64e07c7c76c9ddb1/tap_linkedin_ads/sync.py#L405-L417), which [is based on the bookmark](https://github.com/dreamdata-io/tap-linkedin-ads/blob/c102e151b060c1f5cbbe37ac64e07c7c76c9ddb1/tap_linkedin_ads/sync.py#L326-L331), which [falls back to configuration start date](https://github.com/dreamdata-io/tap-linkedin-ads/blob/c102e151b060c1f5cbbe37ac64e07c7c76c9ddb1/tap_linkedin_ads/sync.py#L317), should it not exist in state.

Again, this whole thing is messy, and this is truly a monkey patch.